lib/govuk_index/popularity_worker.rb

```diff
@@ -0,0 +1,34 @@
+module GovukIndex
+  class PopularityWorker < Indexer::BaseWorker
+    BULK_INDEX_TIMEOUT = 60
+    QUEUE_NAME = 'popularity'.freeze
+    sidekiq_options queue: QUEUE_NAME
+
+    def perform(records, destination_index)
+      actions = ElasticsearchProcessor.new(client: GovukIndex::Client.new(timeout: BULK_INDEX_TIMEOUT, index_name: destination_index))
+
+      popularities = retrieve_popularities_for(records)
+      records.each do |record|
+        actions.save(
+          process_record(record, popularities)
+        )
+      end
+
+      actions.commit
+    end
+
+    def process_record(record, popularities)
+      base_path = record['identifier']['_id']
+      OpenStruct.new(
+        identifier: record['identifier'].merge('_version_type' => 'external_gte'),
+        document: record['document'].merge('popularity' => popularities[base_path]),
+      )
+    end
+
+    def retrieve_popularities_for(records)
+      # The first parameter is only used to check that it is not running for the `page-traffic` index.
```

Comment:
Might be less confusing to pass in the actual index name? It's a bit confusing even with the comment, and I could imagine this causing a bug in the future because the PopularityCode assumes that the index name is correct.